applications/nrf_desktop/src/modules/usb_state.c

```diff
@@ -630,11 +581,84 @@ static void verify_report_bm(void)
 #endif
 }
 
+static int get_report_legacy(const struct device *dev, struct usb_setup_packet *setup,
+			     int32_t *len_to_set, uint8_t **data)
+{
+	uint8_t request_value[2];
+
+	sys_put_le16(setup->wValue, request_value);
+
+	uint8_t report_type = request_value[1];
+	uint8_t report_id = request_value[0];
+	uint8_t *buf = *data;
+	size_t size = setup->wLength;
+
+	if (report_id != REPORT_ID_RESERVED) {
+		buf[0] = report_id;
+		buf++;
+		size--;
+	}
+
+	int err = get_report(dev, report_type, report_id, buf, size);
+
+	if (err) {
+		LOG_WRN("get_report_legacy failed (err: %d)", err);
+		LOG_WRN("bmRequestType: %02X bRequest: %02X wValue: %04X wIndex: %04X"
+			" wLength: %04X", setup->bmRequestType, setup->bRequest,
+			setup->wValue, setup->wIndex, setup->wLength);
+		LOG_HEXDUMP_WRN(buf, size, "Get report");
+	} else {
+		*len_to_set = setup->wLength;
+	}
+
+	return err;
+}
+
+static int set_report_legacy(const struct device *dev, struct usb_setup_packet *setup,
+			     int32_t *len, uint8_t **data)
+{
+	uint8_t request_value[2];
+
+	sys_put_le16(setup->wValue, request_value);
+
+	uint8_t report_type = request_value[1];
+	uint8_t report_id = request_value[0];
```

Comment:
common part, consider defining inline function

```
report_setup_value_parse(uint16_t value, uint8_t *report_type, uint8_t *report_id);
```

Reply:
Since it's used only in 2 places and the inline function would return 2 values through pointers, I would prefer to desist from doing it (prefer to keep parsing explicitly here).

Reply:
Done